### PR TITLE
[kiali] add new sections to illustrate non-graph-related features

### PR DESCRIPTION
**Pull request by jmazzitelli**
Add a new section to the Kiali Task doc.

The new section can be viewed here on netlify: https://deploy-preview-5665--preliminary-istio.netlify.com/docs/tasks/observability/kiali/#creating-weighted-routes

### CODE REVIEW

**Comment by jmazzitelli** [comment]
cc @jstickler

File: content/en/docs/tasks/observability/kiali/index.md [hunk 1]
```diff
@@ -184,11 +184,54 @@ $ oc patch clusterrole kiali -p '[{"op":"add", "path":"/rules/-", "value":{"apiG
...
-   {{< image width="80%" link="./kiali-services.png" caption="Example Details" >}}
+1.  To examine the details about the Istio configuration, click on the
+    **Applications**, **Workloads**, and **Services** menu icons on the left menu
+    bar. The following screenshot shows the Bookinfo applications information:
+
+    {{< image width="80%" link="./kiali-services.png" caption="Example Details" >}}
+
+## Creating weighted routes
+
+Create a weighted route using the Kiali weighted route wizard.
```
Comment (JStickler) [inline content/en/docs/tasks/observability/kiali/index.md]:
```suggestion
You can use the Kiali weighted route wizard to define the specific percentage of request traffic to route to two or more workloads.
```

File: content/en/docs/tasks/observability/kiali/index.md [hunk 2]
```diff
@@ -184,11 +184,54 @@ $ oc patch clusterrole kiali -p '[{"op":"add", "path":"/rules/-", "value":{"apiG
...
+This allows you to route a specific percentage of request traffic to your individual workloads.
```
Comment (JStickler) [inline content/en/docs/tasks/observability/kiali/index.md]:
```suggestion

```

File: content/en/docs/tasks/observability/kiali/index.md [hunk 3]
```diff
@@ -184,11 +184,54 @@ $ oc patch clusterrole kiali -p '[{"op":"add", "path":"/rules/-", "value":{"apiG
...
+
+    *   Make sure you have selected `Requests percentage` in the `Edge Labels` drop down menu
+        to see the percentage of traffic routed to each workload.
+
+    *   Make sure you have selected the `Service Nodes` check box in the `Display` drop down menu
+        to view the service nodes in the graph.
+
+    {{< image width="80%" link="./kiali-wiz0-graph-options.png" caption="Bookinfo Graph Options" >}}
+
+1.  Focus on the `ratings` service within the `bookinfo` graph by clicking on the `ratings` service (triangle) node.
```
Comment (JStickler) [inline content/en/docs/tasks/observability/kiali/index.md]: It looks like these first two steps are just viewing the current configuration.  We should probably make it clear that these steps are not required to create a weighted route, but are just to view the current configuration.
One way to do that is to preface the step with (Optional).

File: content/en/docs/tasks/observability/kiali/index.md [hunk 4]
```diff
@@ -184,11 +184,54 @@ $ oc patch clusterrole kiali -p '[{"op":"add", "path":"/rules/-", "value":{"apiG
...
+    Notice the `ratings` service traffic is evenly distributed to the two `ratings` workloads `v1` and `v2`
+    (i.e. 50% of requests are routed to each workload).
```
Comment (JStickler) [inline content/en/docs/tasks/observability/kiali/index.md]: delete "i.e"

File: content/en/docs/tasks/observability/kiali/index.md [hunk 5]
```diff
@@ -184,11 +184,54 @@ $ oc patch clusterrole kiali -p '[{"op":"add", "path":"/rules/-", "value":{"apiG
...
+
+    {{< image width="80%" link="./kiali-wiz1-graph-ratings-percent.png" caption="Graph Showing Percentage of Traffic" >}}
+
+1.  Click the `ratings` link found in the side panel to go to the service view for the `ratings` service.
+
+1.  Select `Create Weighted Routing` from the `Action` drop down menu to pop up the Weighted Routing Wizard.
```
Comment (JStickler) [inline content/en/docs/tasks/observability/kiali/index.md]: From the *Actions* menu, select 'Create Weighted Routing' to access the Weighted Routing Wizard.   

(You have to find the menu before you can make a choice from it, so give that name first.)

File: content/en/docs/tasks/observability/kiali/index.md [hunk 6]
```diff
@@ -184,11 +184,54 @@ $ oc patch clusterrole kiali -p '[{"op":"add", "path":"/rules/-", "value":{"apiG
...
+
+    {{< image width="80%" link="./kiali-wiz2-ratings-service-action-menu.png" caption="Service Action Menu" >}}
+
+1.  Drag the sliders to change the traffic weights. For `ratings-v1`, set it to 10%; for `ratings-v2` set it to 90%.
```
Comment (JStickler) [inline content/en/docs/tasks/observability/kiali/index.md]: Drag the sliders to specify the percentage of traffic to route to each Workload.

File: content/en/docs/tasks/observability/kiali/index.md [hunk 7]
```diff
@@ -184,11 +184,54 @@ $ oc patch clusterrole kiali -p '[{"op":"add", "path":"/rules/-", "value":{"apiG
...
+
+    {{< image width="80%" link="./kiali-wiz3-weighted-routing-wizard.png" caption="Weighted Routing Wizard" >}}
+
+1.  Click the `Create` button to begin routing 90% of all `ratings` service traffic to `ratings-v2`.
```
Comment (JStickler) [inline content/en/docs/tasks/observability/kiali/index.md]: Click the Create button to create the new routing.

File: content/en/docs/tasks/observability/kiali/index.md [hunk 8]
```diff
@@ -184,11 +184,54 @@ $ oc patch clusterrole kiali -p '[{"op":"add", "path":"/rules/-", "value":{"apiG
...
+
+1.  Go back to the `bookinfo` graph.
```
Comment (JStickler) [inline content/en/docs/tasks/observability/kiali/index.md]: Click Graph in the navigation to return to the `bookinfo` graph page.

File: content/en/docs/tasks/observability/kiali/index.md [hunk 9]
```diff
@@ -184,11 +184,54 @@ $ oc patch clusterrole kiali -p '[{"op":"add", "path":"/rules/-", "value":{"apiG
...
+
+1.  After a few minutes you will notice that the traffic percentage will reflect the new traffic route,
```
Comment (JStickler) [inline content/en/docs/tasks/observability/kiali/index.md]: I think we need a step here to generate traffic, don't we?  Otherwise the graph isn't going to reflect their changes.

Comment (jmazzitelli) [reply content/en/docs/tasks/observability/kiali/index.md]: True - but we do tell the user to generate traffic in an earlier step earlier in the document. But if there is a way we can remind them here to "make sure you are still generating traffic into your bookinfo app", I can add that.

File: content/en/docs/tasks/observability/kiali/index.md (hunk 1 above)
Comment (jmazzitelli) [reply content/en/docs/tasks/observability/kiali/index.md]: done

File: content/en/docs/tasks/observability/kiali/index.md (hunk 2 above)
Comment (jmazzitelli) [reply content/en/docs/tasks/observability/kiali/index.md]: done

File: content/en/docs/tasks/observability/kiali/index.md (hunk 3 above)
Comment (jmazzitelli) [reply content/en/docs/tasks/observability/kiali/index.md]: I'm not sure about changing this here. Every step in the Task aren't necessarily documenting all required steps to use a particular feature in general. These are just steps the user is required to do to complete the Task being documented (and a Task in Istio's documentation is not a fully documenting a feature - a Task is just stepping the user through examples and demos - at least, that is how I understand Istio's Task documentation).

In this case, these steps are required to complete the Task even though the user won't necessary do this when (in general) they create routes via the wizard.

In other words, this Task page isn't documenting how, in general, one uses the wizard to create routes. It's just a Task guide to walk the user through examples.

File: content/en/docs/tasks/observability/kiali/index.md (hunk 4 above)
Comment (jmazzitelli) [reply content/en/docs/tasks/observability/kiali/index.md]: done

File: content/en/docs/tasks/observability/kiali/index.md (hunk 5 above)
Comment (jmazzitelli) [reply content/en/docs/tasks/observability/kiali/index.md]: done

File: content/en/docs/tasks/observability/kiali/index.md (hunk 6 above)
Comment (jmazzitelli) [reply content/en/docs/tasks/observability/kiali/index.md]: done

File: content/en/docs/tasks/observability/kiali/index.md (hunk 7 above)
Comment (jmazzitelli) [reply content/en/docs/tasks/observability/kiali/index.md]: done

File: content/en/docs/tasks/observability/kiali/index.md (hunk 8 above)
Comment (jmazzitelli) [reply content/en/docs/tasks/observability/kiali/index.md]: done

File: content/en/docs/tasks/observability/kiali/index.md (hunk 9 above)
Comment (jmazzitelli) [reply content/en/docs/tasks/observability/kiali/index.md]: I added a step to tell the user to generate traffic.

**Comment by jmazzitelli** [comment]
> General - The Style Guide (https://istio.io/about/contribute/style-guide/) says to use bold for UI elements (buttons, menus, etc.).

I changes these to use **bold** as opposed to the previous `format`

File: content/en/docs/tasks/observability/kiali/index.md (hunk 3 above)
Comment (JStickler) [reply content/en/docs/tasks/observability/kiali/index.md]: I was thinking that if someone was consulting the docs to see how to create a route, those steps aren't necessary.  But if this is more of a tutorial than a procedure, then this is fine.

Comment (jmazzitelli) [reply content/en/docs/tasks/observability/kiali/index.md]: > if this is more of a tutorial

THAT'S the word I was looking for -- "tutorial" :) Yes, this Kiali Task is more of a tutorial.

**Comment by jmazzitelli** [comment]
@geeknoid your thoughts/review on this new section of the Kiali task?

File: content/en/docs/tasks/observability/kiali/index.md [hunk 10]
```diff
@@ -184,11 +190,62 @@ $ oc patch clusterrole kiali -p '[{"op":"add", "path":"/rules/-", "value":{"apiG
 
         {{< image width="70%" link="./kiali-service-graph.png" caption="Example Service Graph" >}}
 
-1. To examine the details about the Istio configuration, click on the
-   **Applications**, **Workloads**, and **Services** menu icons on the left menu
-   bar. The following screenshot shows the Bookinfo applications information:
+## Examining Istio configuration
+
+1.  To examine the details about the Istio configuration, click on the
```
Comment (geeknoid) [inline content/en/docs/tasks/observability/kiali/index.md]:
```suggestion
1.  To view detailed information about Istio configuration, click on the
```

File: content/en/docs/tasks/observability/kiali/index.md [hunk 11]
```diff
@@ -184,11 +190,62 @@ $ oc patch clusterrole kiali -p '[{"op":"add", "path":"/rules/-", "value":{"apiG
...
+    **Applications**, **Workloads**, and **Services** menu icons on the left menu
+    bar. The following screenshot shows the Bookinfo applications information:
```
Comment (geeknoid) [inline content/en/docs/tasks/observability/kiali/index.md]:
```suggestion
    bar. The following screenshot shows information for the the Bookinfo application:
```

File: content/en/docs/tasks/observability/kiali/index.md (hunk 10 above)
Comment (jmazzitelli) [reply content/en/docs/tasks/observability/kiali/index.md]: done

File: content/en/docs/tasks/observability/kiali/index.md (hunk 11 above)
Comment (jmazzitelli) [reply content/en/docs/tasks/observability/kiali/index.md]: done (well, not the double "the" but... :)

**Comment by jmazzitelli** [comment]
@geeknoid @JStickler  I added a new section to go over the kiali validation stuff. You might want to peruse that, also.

See: https://deploy-preview-5665--preliminary-istio.netlify.com/docs/tasks/observability/kiali/#validating-istio-configuration

There is really another one I want to add, but might not have time (I want this to go into the 1.4 docs as soon as 1.4 is released), but I want to show the Yaml editor with configuration errors/warnings shown.

But for now, this PR is ready I think (barring any changes you think we need to do).

File: content/en/docs/tasks/observability/kiali/index.md [hunk 12]
```diff
@@ -247,6 +247,40 @@ request traffic to route to two or more workloads.
 
     {{< image width="80%" link="./kiali-wiz4-ratings-weighted-route-90-10.png" caption="90% Ratings Traffic Routed to ratings-v2" >}}
 
+## Validating Istio configuration
+
+Kiali can validate your Istio resources to ensure they follow proper convention and semantics. Any problems detected in the configuration of your Istio resources can be flagged as errors or warnings depending on the severity of the misconfiguration. See the [Kiali validations page](http://kiali.io/documentation/validations/) for the list of all validation checks Kiali performs.
```
Comment (geeknoid) [inline content/en/docs/tasks/observability/kiali/index.md]: convention -> conventions

File: content/en/docs/tasks/observability/kiali/index.md [hunk 13]
```diff
@@ -247,6 +247,40 @@ request traffic to route to two or more workloads.
...
+
+Force a misconfiguration of a service port name to see how Kiali reports a validation error.
```
Comment (geeknoid) [inline content/en/docs/tasks/observability/kiali/index.md]: How about adding:

{{< idea >}}
Istio 1.4 introduces [`istioctl analyze`](/docs/ops/diagnostic-tools/istioctl-analyze/) which lets you perform similar analysis in a way that can be used in a CI pipeline.
{{< /idea >}}

File: content/en/docs/tasks/observability/kiali/index.md (hunk 12 above)
Comment (jmazzitelli) [reply content/en/docs/tasks/observability/kiali/index.md]: done

File: content/en/docs/tasks/observability/kiali/index.md (hunk 13 above)
Comment (jmazzitelli) [reply content/en/docs/tasks/observability/kiali/index.md]: done
